### DISCUSSION TITLE
Uploading object to S3

### DISCUSSION THREAD

**Discussion by quell01102022** [miurahr/py7zr#621] [Q&A]
Hey, so i can see i can use extract and extractall to write to filesystem. But i need an extracted object to write to S3 Bucket like here:
```
with py7zr.SevenZipFile("file.7z", "r") as archive:
    for image in archive.namelist():
        s3resource.meta.client.upload_fileobj( archive.open(image), Bucket="bucket", Key="key")



# in zipfile, i can use archive.open() for example, how can i achieve same here?
```

**Comment by Ravencentric** [comment]
Use [`.read()`](https://py7zr.readthedocs.io/en/latest/api.html#py7zr.SevenZipFile.read)

**Comment by miurahr** [comment]
I plan to drop `read` method that lead users wrong way.

please check a new way to stream content to user without consuming memory.
https://github.com/miurahr/py7zr/pull/620
